adapters/coap-vertx-base/src/main/java/org/eclipse/hono/adapter/coap/impl/ConfigBasedCoapEndpointFactory.java

```diff
@@ -326,6 +326,7 @@ private NetworkConfig newDefaultNetworkConfig() {
         dtlsConfig.setAdvancedPskStore(pskStore);
         dtlsConfig.setRetransmissionTimeout(config.getDtlsRetransmissionTimeout());
         dtlsConfig.setMaxConnections(networkConfig.getInt(Keys.MAX_ACTIVE_PEERS));
+        dtlsConfig.setSniEnabled(true);
```

Comment:
I've to check. I think that also changes the used PSK names ... 
Maybe we need additional configuration in Californium 3.

Reply:
I checked it in the source:
It will change the [PreSharedKeyIdentity - Name](https://github.com/eclipse/californium/blob/master/element-connector/src/main/java/org/eclipse/californium/elements/auth/PreSharedKeyIdentity.java).
I don't see, that this is used (via `Principal.getName()), so it should not harm.
For the device lookup itself, it's left to the `DeviceRegistryBasedPskStore`, which doesn't use it  for now.
So, from checking the sources, that should work.